src/main/java/com/pracownia/vanet/view/Simulation.java

```diff
@@ -93,9 +90,22 @@ private void buildRoads() {
         crossRoads.add(new CrossRoad(new Point(800.0, 400.0), roads.get(3), roads.get(5)));
         crossRoads.add(new CrossRoad(new Point(800.0, 600.0), roads.get(3), roads.get(6)));
 
+        devices.add(new SIN(carCounter.getAndIncrement(), new Point(500.0, 400.0), 50.0));
         devices.add(new RoadSide(carCounter.getAndIncrement(), new Point(480.0, 210.0), 50.0));
         devices.add(new RoadSide(carCounter.getAndIncrement(), new Point(260.0, 610.0), 50.0));
         devices.add(new RoadSide(carCounter.getAndIncrement(), new Point(480.0, 610.0), 50.0));
+        if (devices.stream().findFirst().get() instanceof  SIN) {
```

Comment:
Dlaczego szukacie tu obiektu, który tworzony jest trzy linijki wyżej? Poza tym ta implementacja jest bardzo wrażliwa na kolejność elementów jeśli inna grupa doda swój device nad waszym SINem to wasz kod w tym momencie już się nie uruchomi.

Reply:
Poprawione. Dodane metody do wyszukiwania SINa.